### PR TITLE
feat: LanguageSwitcher helper class

### PR DESCRIPTION
**Pull request by rslibed**

### CODE REVIEW

File: src/structuralFunctionality/language-switcher/LanguageSwitcher.tsx [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+        handleKey
+      );
+    }
+  }
+
+  private async _languageSwitcherConfigCallback(widgetProps: esriWidgetProps): Promise<void> {
+    const expand = widgetProps?.view?.ui?.find("esri-language-switcher") as Expand;
+    expand.expandIcon = this.configurationSettings.languageSwitcherConfig.icon;
+    this.configurationSettings.languageSwitcherConfig =
+      this.configurationSettings.languageSwitcherConfig;
```
[the remark this answers is not in the document]
Comment (kellyhutchins) [reply src/structuralFunctionality/language-switcher/LanguageSwitcher.tsx:247]: Is this needed? Looks like the code scanning caught it as an issue

File: src/structuralFunctionality/language-switcher/LanguageSwitcher.tsx [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+    await this._refresh();
+    this._setLanguageSwitcherUI(this.base.config, this.configurationSettings);
+  }
+
+  private _isWithinConfigurationExperience(): boolean {
```
Comment (kellyhutchins) [inline src/structuralFunctionality/language-switcher/LanguageSwitcher.tsx]: Looks like we have this in app base in the TCL. Might make sense to move isWIthinConfigurationExperience to a utils class?

**Review by kellyhutchins — APPROVED** [review]

File: src/structuralFunctionality/language-switcher/LanguageSwitcher.tsx (hunk 2 above)
Comment (rslibed) [reply src/structuralFunctionality/language-switcher/LanguageSwitcher.tsx]: Yep makes sense - just moved it over to a separate util function

File: src/structuralFunctionality/language-switcher/LanguageSwitcher.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (rslibed) [reply src/structuralFunctionality/language-switcher/LanguageSwitcher.tsx]: Done

**Pull request merged**